src/pybind/mgr/cephadm/inventory.py

```diff
@@ -174,6 +174,9 @@ def find(self, service_name: Optional[str] = None) -> List[ServiceSpec]:
             service_name, specs))
         return specs
 
+    def get_created(self, spec: ServiceSpec) -> Optional[datetime.datetime]:
+        return self.spec_created.get(spec.service_name())
```

Comment:
this should always work. If we really want to verify the integrity here, what about

```suggestion
    def get_created(self, spec: ServiceSpec) -> Optional[datetime.datetime]:
        if spec.service_name() not in self.spec_created
            self.save(spec.service_name())
        return self.spec_created[spec.service_name()])
```